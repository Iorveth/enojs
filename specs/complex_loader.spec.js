const { parse } = require('../eno.js');

const input = `
bad coordinates: Mickey Mouse
good coordinates: 48.211180, 16.371514
`.trim();

const loadCoordinates = ({ name, value }) => {
  const split = value.split(',');

  if(split.length < 2) {
    throw `${name} should have a lat and a lng!`;
  }

  return {
    lat: parseFloat(split[0]),
    lng: parseFloat(split[1])
  };
};

describe('Complex loader', () => {
  const doc = parse(input);

  it('throws with bad input', () => {
    expect(() => doc.field('bad coordinates', loadCoordinates)).toThrowErrorMatchingSnapshot();
  });

  it('works with good input', () => {
    expect(doc.field('good coordinates', loadCoordinates)).toEqual({ lat: 48.211180, lng: 16.371514 });
  });
});
