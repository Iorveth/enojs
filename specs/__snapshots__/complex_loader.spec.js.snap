// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Complex loader throws with bad input 1`] = `
"bad coordinates should have a lat and a lng!

   Line | Content
 >    1 | bad coordinates: Mickey Mouse
      2 | good coordinates: 48.211180, 16.371514
"
`;
